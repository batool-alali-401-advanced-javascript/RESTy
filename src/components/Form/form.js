import React from "react";
import './form.scss';


class Main extends React.Component {
    constructor(props) {
      super(props);
      this.state = { 
          url: '',
          method:'',
          full:'',
     };
    }

    handleChange = (e) =>{
        const url = e.target.value;
        console.log(url);
        console.log(this.state.url);
        this.setState({ url: url });

    };

    handleClickGo = (e) => {
        this.setState({full: `${this.state.method}     ${this.state.url} `})
    };

    handleClickGet = (e) => {
        this.setState({ method: 'get' });
    };
    handleClickPost = (e) => {
        this.setState({ method: 'post' });
    };
    handleClickPut = (e) => {
        this.setState({ method: 'put' });
    };
    handleClickDelete = (e) => {
        this.setState({ method: 'delete' });
    };
    
  render() {
    return (
        <main className="mainForm">
            <div className='form'>
                <label className="urlLabel">URL:</label>
                <input type="text" className='urlInput' onChange={this.handleChange} />
                <button className='method' onClick={this.handleClickGo}>Go!</button>
                <br></br>
                <button className='method' onClick={this.handleClickGet}>GET</button>
                <button className='method' onClick={this.handleClickPost}>POST</button>
                <button className='method' onClick={this.handleClickPut}>PUT</button>
                <button className='method' onClick={this.handleClickDelete}>DELETE</button>
            </div>
       
        <div className='fullMethod'>{this.state.full}</div>



      </main>
    );
  }
}


  
  export default Main;
