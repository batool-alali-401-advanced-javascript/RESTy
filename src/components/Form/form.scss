.mainForm{
    & .form{
        text-align: center;
        .method{
            margin: 5px;
        }
    }
     
    
}

.fullMethod{
    width: 90%;
    height: 40%;
    min-height: 200px;
    min-width: 1200px;
    border-style: inset;
    margin: 10px;
}