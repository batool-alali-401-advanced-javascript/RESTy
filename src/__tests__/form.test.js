import React from 'react';
import { shallow, mount } from 'enzyme';
import renderer from 'react-test-renderer';
import Form from '../components/Form/form.js';


describe('<Form/>', () => {
    it('is alive at application start', () => {
      const form = shallow(<Form />);
      expect(form.find('main').exists()).toBeTruthy();
    });
    it('changes method on click on GET', () => {
      const form = mount(<Form />);
      const button = form.find('#get');
      button.simulate('click');
      expect(form.state('method')).toBe('get');
    });
    it('changes method on click on POST', () => {
        const form = mount(<Form />);
        const button = form.find('#post');
        button.simulate('click');
        expect(form.state('method')).toBe('post');
      });
      it('changes method on click on PUT', () => {
        const form = mount(<Form />);
        const button = form.find('#put');
        button.simulate('click');
        expect(form.state('method')).toBe('put');
      });
      it('changes method on click on DELETE', () => {
        const form = mount(<Form />);
        const button = form.find('#delete');
        button.simulate('click');
        expect(form.state('method')).toBe('delete');
      });
      it('changes URL on Change the input', () => {
        const form = mount(<Form />);
        const input = form.find('.urlInput');
        input.simulate('change', { target: { value: 'http://localhost:3000' } });
        expect(form.state('url')).toBe('http://localhost:3000');
      });
      it('changes the full request on click Go!', () => {
        const form = mount(<Form />);
        const go = form.find('#go');
        const input = form.find('.urlInput');
        input.simulate('change', { target: { value: 'http://localhost:3000' } });
        const button = form.find('#delete');
        button.simulate('click');
        go.simulate('click');
        expect(form.state('full')).toBe("delete     http://localhost:3000 ");
      });


  
    it('Renders correctly', () => {
      const tree = renderer.create(<Form />).toJSON();
      expect(tree).toMatchSnapshot();
    });
  });