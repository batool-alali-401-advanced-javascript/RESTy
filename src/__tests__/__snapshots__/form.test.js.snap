// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Form/> Renders correctly 1`] = `
<main
  className="mainForm"
>
  <div
    className="form"
  >
    <label
      className="urlLabel"
    >
      URL:
    </label>
    <input
      className="urlInput"
      onChange={[Function]}
      type="text"
    />
    <button
      className="method"
      id="go"
      onClick={[Function]}
    >
      Go!
    </button>
    <br />
    <button
      className="method"
      id="get"
      onClick={[Function]}
    >
      GET
    </button>
    <button
      className="method"
      id="post"
      onClick={[Function]}
    >
      POST
    </button>
    <button
      className="method"
      id="put"
      onClick={[Function]}
    >
      PUT
    </button>
    <button
      className="method"
      id="delete"
      onClick={[Function]}
    >
      DELETE
    </button>
  </div>
  <div
    className="fullMethod"
  >
     
  </div>
</main>
`;
